core/src/database/DatabaseBackend.cpp

```diff
@@ -40,6 +41,10 @@ namespace ledger {
             return std::make_shared<SQLite3Backend>();
         }
 
+        std::shared_ptr<api::DatabaseBackend> api::DatabaseBackend::getPostgreSQLBackend(int32_t connectionPoolSize) {
```

Comment:
I think we need a way to get the `PostgreSQLBackend` directly. The problem with this code is that even though it’s fine for interface code (they will use the `api::DatabaseBackend` API) because upcast are free in C++, it’s not the case whenever we need the concrete type, resulting in a downcast and hence a dynamic cast.